backends/ebpf/codeGen.cpp

```diff
@@ -50,7 +51,10 @@ bool CodeGenInspector::preorder(const IR::Declaration_Variable* decl) {
 bool CodeGenInspector::preorder(const IR::Operation_Binary* b) {
     widthCheck(b);
     int prec = expressionPrecedence;
-    bool useParens = prec > b->getPrecedence();
+    bool useParens = false;
+    if ((prec > b->getPrecedence()) ||
```

Comment:
I think that what you want here is to check whether the parent is an if statement:
`useParents = getParent<IR::IfStatement>() == nullptr;`
This will use parentheses for everything but not for the expression in an if condition.